packages/astro/package.json

```diff
@@ -147,6 +147,7 @@
     "strip-ansi": "^7.0.1",
     "supports-esm": "^1.0.0",
     "tsconfig-resolver": "^3.0.1",
+    "typescript": "*",
```

Comment:
is this safe?

Reply:
*yes and no*, if you don't have TypeScript installed in your project, then updating Astro might update the TypeScript version which might break the type checking if TypeScript introduce a breaking change

If you do have TypeScript installed in your project, it should normally just use the workspace's version and so it won't break

Reply:
Curious why this was done in this PR. Can we not update the language server without this?

Reply:
Starting from version 0.24.0, the language server doesn't bundle TypeScript anymore (to allow users to control the version, like they would for .ts files and other reasons). 

So yeah, can't upgrade the language server without breaking `astro check`, so we bundle TypeScript into Astro itself instead. In the next update where we can do breaking changes we'll remove this and instead offer a message that says to install TypeScript to use `astro check`